File: src/main/scala/com/github/mauricio/postgresql/parsers/ErrorParser.scala
package com.github.mauricio.postgresql.parsers

import com.github.mauricio.postgresql.messages.backend.{ErrorMessage, Message}
import java.nio.charset.Charset

class ErrorParser( charset : Charset ) extends InformationParser(charset) {

  def createMessage(fields: Map[Char, String]): Message = new ErrorMessage(fields)

}

File: postgresql-async/src/main/java/com/github/mauricio/async/db/postgresql/parsers/ErrorParser.kt
package com.github.mauricio.async.db.postgresql.parsers

import com.github.mauricio.async.db.postgresql.messages.backend.ErrorMessage
import com.github.mauricio.async.db.postgresql.messages.backend.ServerMessage
import java.nio.charset.Charset


class ErrorParser(charset: Charset) : InformationParser(charset) {

  override fun createMessage(fields: Map<Char, String>): ServerMessage = ErrorMessage(fields)

}
